centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala

```diff
@@ -107,7 +107,16 @@ object CentaurCromwellClient extends StrictLogging {
     Try(Await.result(successOrFailure, CentaurConfig.sendReceiveTimeout)).isSuccess
   }
 
-  def metadata(workflow: SubmittedWorkflow, args: Option[Map[String, List[String]]] = defaultMetadataArgs): IO[WorkflowMetadata] = metadataWithId(workflow.id, args)
+  def metadata(workflow: SubmittedWorkflow,
+               args: Option[Map[String, List[String]]] = defaultMetadataArgs,
+               expandSubworkflows: Boolean = false,
+               archived: Option[Boolean] = None): IO[WorkflowMetadata] = {
+    val mandatoryArgs = Map("expandSubWorkflows" -> List(expandSubworkflows.toString))
+    val metadataSourceOverrideOpt = archived map { isArchived =>
+      "metadataSource" -> List(if (isArchived) "Archived" else "Unarchived")
+    }
+    metadataWithId(workflow.id, Option(mandatoryArgs ++ metadataSourceOverrideOpt.toMap ++ args.getOrElse(Map.empty)))
```

Comment:
TOL: in this order, any items in `mandatoryArgs` would be overwritten by `args` if they happened to collide. Is that expected and intended?